WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -231,15 +274,100 @@ public void hideLoadingProgress() {
         }
     }
 
+    /*
+     * use this to add custom control/component views to your toolbar. Components are added on the right side of the
+      * toolbar */
+    public void addToolbarCustomControl(View v, OnClickListener clickListener){
+        if (v != null){
+            mCustomComponentsContainer.addView(v);
+            if (clickListener != null){
+                v.setOnClickListener(clickListener);
+            }
+        }
+    }
+
+    public void setToolbarBackgroundColor(int color){
+        mToolbar.setBackgroundColor(color);
+    }
+
+    public void setToolbarSpinnerTextColor(int color){
+        mSpinnerTextColor = color;
+    }
+
+    public void setToolbarSpinnerDrawable(int drawableResId){
+        mSpinnerDrawableRight = drawableResId;
+    }
+
+    public void setToolbarLeftPadding(int paddingLeft){
+        mToolbar.setPadding(paddingLeft,
+                mToolbar.getPaddingTop(),
+                mToolbar.getPaddingRight(),
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarRightPadding(int paddingRight){
+        mToolbar.setPadding(
+                mToolbar.getPaddingLeft(),
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void setToolbarLeftAndRightPadding(int paddingLeft, int paddingRight){
+        mToolbar.setPadding(
+                paddingLeft,
+                mToolbar.getPaddingTop(),
+                paddingRight,
+                mToolbar.getPaddingBottom());
+    }
+
+    public void scrollRecycleViewToPosition(int position) {
+        if (mRecyclerView == null) return;
+
+        mRecyclerView.scrollToPosition(position);
+    }
+
+    public int getCurrentPosition() {
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            return ((LinearLayoutManager) mRecyclerView.getLayoutManager()).findFirstVisibleItemPosition();
+        } else {
+            return -1;
+        }
+    }
+
+    public void smoothScrollToPosition(int position){
+        if (mRecyclerView != null && mRecyclerView.getLayoutManager() != null) {
+            mRecyclerView.getLayoutManager().smoothScrollToPosition(mRecyclerView, null, position);
+        }
+    }
+
+    /*
+    * use this to access the internal RecyclerView if you need to, for example, add items decoration
+    * */
+    public RecyclerView getInternalRecyclerView(){
```

Comment:
This seems a bit ugly, requiring the caller to know there's an "internal recycler view." Since this is only used for item decorations right now, perhaps just create an `addItemDecoration()` method in this class instead?